src/metric/ackbag.rs

```diff
@@ -0,0 +1,104 @@
+use std::sync::{Arc, Mutex};
+use std::{thread, time};
+use util;
+use uuid::Uuid;
+
+#[derive(Clone)]
+pub struct SyncAckProps {
+    acks_received: usize
+}
+
+impl SyncAckProps {
+    pub fn new() -> Self {
+        SyncAckProps {
+            acks_received: 0,
+        }
+    }
+
+    pub fn ack(&mut self) {
+        self.acks_received.saturating_add(1);
```

Comment:
This function could be `&self`. Or do you mean to do `self.acks_received = self.acks_received.saturating_add(1);` ?